src/autolabel/tasks/multilabel_classification.py

```diff
@@ -0,0 +1,200 @@
+
+        # prepare task guideline
+        labels_list = self.config.labels_list()
+        num_labels = len(labels_list)
+        fmt_task_guidelines = self.task_guidelines.format(
+            num_labels=num_labels, labels="\n".join(labels_list)
+        )
+
+        # prepare seed examples
+        example_template = self.config.example_template()
+        label_column = self.config.label_column()
+        fmt_examples = []
+        for eg in examples:
+            eg_copy = eg.copy()
+            # If chain of thought is enabled
+            if label_column and self.config.chain_of_thought():
+                eg_copy[label_column] = json.dumps({label_column: eg[label_column]})
+            fmt_examples.append(example_template.format_map(defaultdict(str, eg_copy)))
+
+        # populate the current example in the prompt
+        if label_column:
+            input[label_column] = ""
+
+        # populate the explanation column with empty string for current example
+        explanation_column = self.config.explanation_column()
+        if explanation_column:
+            input[explanation_column] = ""
+
+        # populate the current example in the prompt
+        current_example = example_template.format_map(defaultdict(str, input))
+
+        if self._is_few_shot_mode():
+            return self.prompt_template.format(
+                task_guidelines=fmt_task_guidelines,
+                output_guidelines=self.output_guidelines,
+                seed_examples="\n\n".join(fmt_examples),
+                current_example=current_example,
+            )
+        else:
+            return self.prompt_template.format(
+                task_guidelines=fmt_task_guidelines,
+                output_guidelines=self.output_guidelines,
+                current_example=current_example,
+            )
+
+    def get_explanation_prompt(self, example: Dict) -> str:
+        pt = PromptTemplate(
+            input_variables=get_format_variables(self.GENERATE_EXPLANATION_PROMPT),
+            template=self.GENERATE_EXPLANATION_PROMPT,
+        )
+
+        # prepare task guideline
+        labels_list = self.config.labels_list()
+        num_labels = len(labels_list)
+        fmt_task_guidelines = self.task_guidelines.format(
+            num_labels=num_labels, labels="\n".join(labels_list)
+        )
+
+        # prepare labeled example
+        example_template = self.config.example_template()
+        fmt_example = example_template.format_map(defaultdict(str, example))
+
+        return pt.format(
+            task_guidelines=fmt_task_guidelines,
+            labeled_example=fmt_example,
+        )
+
+    def auroc_score_labels(
+        self, gt_labels, llm_labels
+    ) -> Tuple[List[int], List[float]]:
+        labels = []
+        confidences = []
+        for index, llm_label in enumerate(llm_labels):
+            labels.append(llm_label.label.lower() == gt_labels[index].lower())
+            confidences.append(llm_label.confidence_score)
```

Comment:
Confidence here will be the average of log probs of all the tokens in the llm labels. 

We might instead want to define confidence individually for each each label in the list. Not sure if that is in the scope of this PR - if not, let's open a ticket for doing this as a follow up.

Reply:
Implementing this will require some revamp on the confidence calculation side. We probably want to now move confidence calculation into each task (and the parent class can implement the default confidence calculator)

Reply:
Just created #406 for this